server/src/main/java/org/elasticsearch/cluster/metadata/Metadata.java

```diff
@@ -696,89 +704,98 @@ public Iterator<? extends ToXContent> toXContentChunked(ToXContent.Params p) {
+        }
+    }
 
-            final var iterators = Iterators.concat(start, Iterators.single((builder, params) -> {
-                builder.field("cluster_uuid", clusterUUID);
-                builder.field("cluster_uuid_committed", clusterUUIDCommitted);
-                builder.startObject("cluster_coordination");
-                coordinationMetadata().toXContent(builder, params);
-                return builder.endObject();
-            }),
-                persistentSettings,
-                project.toXContentChunked(p),
+    private Iterator<? extends ToXContent> toXContentChunkedWithMultiProjectFormat(
+        ToXContent.Params p,
+        XContentContext context,
+        Iterator<? extends ToXContent> start,
+        Iterator<? extends ToXContent> clusterCoordination,
+        Iterator<? extends ToXContent> persistentSettings
+    ) {
+        return Iterators.concat(
+            start,
+            clusterCoordination,
+            persistentSettings,
+            ChunkedToXContentHelper.array(
+                "projects",
+                Iterators.flatMap(
+                    projectMetadata.entrySet().iterator(),
+                    e -> Iterators.concat(
+                        ChunkedToXContentHelper.startObject(),
+                        Iterators.single((builder, params) -> builder.field("id", e.getKey())),
+                        e.getValue().toXContentChunked(p),
+                        ChunkedToXContentHelper.endObject()
+                    )
+                )
+            ),
+            Iterators.flatMap(
+                customs.entrySet().iterator(),
+                entry -> entry.getValue().context().contains(context)
+                    ? ChunkedToXContentHelper.object(entry.getKey(), entry.getValue().toXContentChunked(p))
+                    : Collections.emptyIterator()
+            ),
+            ChunkedToXContentHelper.object("reserved_state", reservedStateMetadata().values().iterator()),
+            ChunkedToXContentHelper.endObject()
+        );
+    }
+
+    private Iterator<? extends ToXContent> toXContentChunkedWithSingleProjectFormat(
+        ToXContent.Params p,
+        XContentContext context,
+        Iterator<? extends ToXContent> start,
+        Iterator<? extends ToXContent> clusterCoordination,
+        Iterator<? extends ToXContent> persistentSettings
+    ) {
+        if (projectMetadata.size() != 1) {
+            throw new MultiProjectPendingException("There are multiple projects " + projectMetadata.keySet());
+        }
+        // Need to rethink what to do here. This might be right, but maybe not...
+        @FixForMultiProject
+        final ProjectMetadata project = projectMetadata.values().iterator().next();
+
+        // need to combine reserved state together into a single block so we don't get duplicate keys
+        // and not include it in the project xcontent output (through the lack of multi-project params)
+        // use a tree map so the order is deterministic
```

Comment:
We don't need the tree map in the MP version?

Reply:
I actually have no idea. This is existing code. I thought the order was relied on tests. But I could not find any evidence.